src/CommentSold/Api/ShopClient.php

```diff
@@ -0,0 +1,331 @@
+    }
+
+    /**
+     * Returns a paginated list of customers
+     */
+    public function getCustomers(int $page = 1, int $perPage = Rest::PER_PAGE)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers?page={$page}&perPage={$perPage}");
+    }
+
+    /**
+     * Add a new customer
+     */
+    public function createCustomer(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers", $payload);
+    }
+
+    /**
+     * Returns a customer for the given CommentSold customer id
+     */
+    public function getCustomer(int $customerId)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers/{$customerId}");
+    }
+
+    /**
+     * Update an existing customer
+     */
+    public function updateCustomer(int $customerId, array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers/{$customerId}", $payload);
+    }
+
+    /**
+     * Returns a paginated list of customers filtered by the search term
+     */
+    public function searchCustomers(string $searchTerm, int $page = 1, int $perPage = Rest::PER_PAGE)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/customers/search?page={$page}&perPage={$perPage}", ['term' => $searchTerm]);
+    }
+
+    /**
+     * Returns a customer for the given external customer id
+     */
+    public function getCustomerByExternalId(string $customerId)
+    {
+        return $this->restClient->get($this->shopToken, "{$this->shopId}/customers/externalId/{$customerId}");
+    }
+
+    /**
+     * Updates the address for the shop
+     */
+    public function updateShop(array $payload)
+    {
+        return $this->restClient->patch($this->shopToken, "{$this->shopId}/shop", $payload);
+    }
+
+    /**
+     * Update the products to have the provided category. The existing category for the products will be overwritten.
+     */
+    public function updateProductCategories(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/product_categories", $payload);
+    }
+
+    /**
+     * Start allocations for dropship products
+     */
+    public function startDropshipAllocation(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/dropshipping/allocations", $payload);
+    }
+
+    /**
+     * Restock allocations for dropship products
+     */
+    public function restockDropshipAllocation(array $payload)
+    {
+        return $this->restClient->put($this->shopToken, "{$this->shopId}/dropshipping/allocations", $payload);
+    }
+
+    /**
+     * Calculate the estimated shipping price for a group of variants
+     */
+    public function estimateShipping(array $payload)
+    {
+        return $this->restClient->post($this->shopToken, "{$this->shopId}/shipping_calculator", $payload);
+    }
+}
```

Comment:
I think it would make sense to break the client app for each module (ProductClient, ShippingClient, etc) rather than a giant list of all endpoints.